sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -1392,25 +1407,28 @@ static void _event_cb(netdev_t *dev, netdev_event_t event)
     }
     else {
         DEBUG("gnrc_netif: event triggered -> %i\n", event);
+        gnrc_pktsnip_t *pkt = NULL;
         switch (event) {
-            case NETDEV_EVENT_RX_COMPLETE: {
-                    gnrc_pktsnip_t *pkt = netif->ops->recv(netif);
-
-                    if (pkt) {
-                        _pass_on_packet(pkt);
-                    }
+            case NETDEV_EVENT_RX_COMPLETE:
+                pkt = netif->ops->recv(netif);
+                if (pkt) {
+#ifdef MODULE_NETSTATS_L2
+                    netif->stats.rx_count++;
+                    netif->stats.rx_bytes += pkt->size;
+#endif
```

Comment:
The current gnrc L2 specific code removes the L2 header from the `pkt`. Using the size of `pkt` here measures 14 bytes too few for Ethernet for each received packet.

I think this would have to be moved to the layer specific `recv` implementations.